src/NodaTime.Test/Calendars/BclWeekYearRuleTest.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright 2016 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NUnit.Framework;
+using System;
+using System.Globalization;
+using static System.Globalization.CalendarWeekRule;
+using static System.DayOfWeek;
+using NodaTime.Calendars;
+
+namespace NodaTime.Test.Calendars
+{
+    public class BclWeekYearRuleTest
+    {
+        /// <summary>
+        /// For each calendar and rule combination, check everything we can about every date
+        /// from mid December to mid January around each year between 2016 and 2046.
+        /// (For non-Gregorian calendars, the rough equivalent is used...)
+        /// That should give us plenty of coverage.
+        /// </summary>
+        [Test]
+        [Combinatorial]
+        public void BclEquivalence(
+            [ValueSource(typeof(BclCalendars), nameof(BclCalendars.MappedCalendars))] Calendar calendar,
+            [Values(FirstDay, FirstFourDayWeek, FirstFullWeek)] CalendarWeekRule bclRule,
+            [Values(Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday)]  DayOfWeek firstDayOfWeek)
+        {
+            var nodaCalendar = BclCalendars.CalendarSystemForCalendar(calendar);
+            var nodaRule = BclWeekYearRule.GetRule(bclRule, firstDayOfWeek);
+            var startYear = new LocalDate(2016, 1, 1).WithCalendar(nodaCalendar).Year;
+
+            for (int year = startYear; year < startYear + 30; year++)
+            {
+                var startDate = new LocalDate(year, 1, 1, nodaCalendar).PlusDays(-15);
+                for (int day = 0; day < 30; day++)
+                {
+                    var date = startDate.PlusDays(day);
+                    var bclDate = date.ToDateTimeUnspecified();
+                    var bclWeek = calendar.GetWeekOfYear(bclDate, bclRule, firstDayOfWeek);
+                    // Weird... the BCL doesn't have a way of finding out which week-year we're in.
+                    var bclWeekYear = bclWeek < 10 ? year : year - 1;
```

Comment:
Probably worth adding to the comment: we start at start-of-year-minus-15-days (which is easy to miss, btw), and so 'small' week-of-year values mean that we're in the current year, whereas larger ones mean that we're in the prior year.